### ISSUE TITLE
Bug: aXe accessibility audit failures on the homepage

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [microsoft/TypeScript-Website#1005]
**Improvement Idea:**

I love that you did an a11y audit as part of the redesign of the site! It's mostly working great over screenreader, yay! 🙌 

However, there are a few little slightly incorrect behaviors on the homepage as caught by the [aXe developer tools](https://www.deque.com/axe/browser-extensions/):

* Improper ARIA role for `a#homepage-logo`: there's already a `<header>` (which here defaults to the `banner` role) wrapping that link. `banner` should be reserved for the site header content that is common across the site generally at the top of every page. [[docs](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/Banner_role)]
* Headings out of order: It can be disorienting for users who navigate by jumping across headings to see an `h3` immediately following an `h1`, as headings are generally meant to increase in rank by 1 at a time. [[docs](https://www.w3.org/WAI/tutorials/page-structure/headings)]

**Why:**

None of these are ship-blocking for, say, a non-sighted / keyboard-only user to use the site. But they are quality of life improvements that can make it better.

I'd be happy to send a PR to resolve these!

**Comment by orta** [comment]
Works for me, aXe isn't on our set of tools to validate (OS screenreaders and NVDA) but would rather cover more cases

I think the `What is TypeScript?` could be a H2, then that should settle the 2nd
